### PR TITLE
20). Add topic files 51 to 54

### PR DESCRIPTION
**Pull request by jameshkramer**
@mairaw @BillWagner
This PR is for user story #1576037.

It adds four article files, makes corresponding deletions from the C# and VB folders, and adds redirection entries. There are also fixes to two past articles.

Please merge #17688 before merging this.

cc: @tfosmark @marteeleigh

Contributes to #4728

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This LGTM @jameshkramer 

I'll wait to :shipit: until @mairaw has a chance to review.

**Comment by mairaw** [comment]
@jameshkramer can you review the merge conflicts here?

**Comment by jameshkramer** [comment]
@mairaw   @BillWagner 

I fixed the merge conflicts. Thanks for reviewing so quickly!

**Comment by jameshkramer** [comment]
@mairaw @BillWagner 

Is there a reason why this hasn't been merged?

**Comment by mairaw** [comment]
I haven’t had a chance to review yet. Will do it tomorrow

**Comment by jameshkramer** [comment]
@mairaw, sorry, I thought your merge indicated you had looked at it.

File: docs/standard/linq/xdocument-class-overview.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: XDocument class overview
-description: The LINQ to XML XDocument class contains the information necessary for a valid XML document. In many cases you do not require the functionality of an XDocument object and can use an XElement object instead.
+description: The LINQ to XML XDocument class contains the information necessary for a valid XML document. In many cases you don't require the functionality of an XDocument object and can use an XElement object instead.
```
Comment (mairaw) [inline docs/standard/linq/xdocument-class-overview.md:3]:
```suggestion
description: The LINQ to XML XDocument class contains the information necessary for a valid XML document. In many cases, you don't need the functionality of an XDocument object and can use an XElement object instead.
```

**Review by mairaw — CHANGES_REQUESTED** [review]
I'm not seeing the redirection file updates and the files being deleted to introduce filter-optional-element.md, find-all-nodes-namespace.md, sort-elements.md, and write-queries-complex-filtering.md.

**Comment by jameshkramer** [comment]
Maira, you were right - I didn't do that work. Sorry for the bother. I think it's fixed now.

File: docs/standard/linq/filter-optional-element.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+## See also
+
+- <xref:System.Xml.Linq.XElement.Attribute%2A?displayProperty=nameWithType>
+- <xref:System.Xml.Linq.XContainer.Elements%2A?displayProperty=nameWithType>
+- <xref:System.Xml.Linq.Extensions.Elements%2A?displayProperty=nameWithType>
+- [Standard Query Operators Overview (C#)](../../csharp/programming-guide/concepts/linq/standard-query-operators-overview.md)
+- [Projection Operations (C#)](../../csharp/programming-guide/concepts/linq/projection-operations.md)
+- [Basic Queries (LINQ to XML) (Visual Basic)](../../visual-basic/programming-guide/concepts/linq/basic-queries-linq-to-xml.md)
+- [XML Child Axis Property](../../visual-basic/language-reference/xml-axis/xml-child-axis-property.md)
```
Comment (mairaw) [inline docs/standard/linq/filter-optional-element.md:180]:
```suggestion
- [XML Child Axis Property (Visual Basic)](../../visual-basic/language-reference/xml-axis/xml-child-axis-property.md)
```

File: docs/standard/linq/filter-optional-element.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+- [XML Attribute Axis Property](../../visual-basic/language-reference/xml-axis/xml-attribute-axis-property.md)
```
Comment (mairaw) [inline docs/standard/linq/filter-optional-element.md:181]:
```suggestion
- [XML Attribute Axis Property (Visual Basic)](../../visual-basic/language-reference/xml-axis/xml-attribute-axis-property.md)
```

File: docs/standard/linq/sort-elements.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
+---
+title: How to sort elements - LINQ to XML
+description: Wite a query that sorts its results
```
Comment (mairaw) [inline docs/standard/linq/sort-elements.md:3]:
```suggestion
description: Write a query that sorts its results.
```

File: docs/standard/linq/sort-elements.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+24.50
+29.00
+66.00
+89.99
+```
+
+## See also
+
+- [Sorting Data (C#)](../../csharp/programming-guide/concepts/linq/sorting-data.md)
+- [Sorting Data](../../visual-basic/programming-guide/concepts/linq/sorting-data.md)
```
Comment (mairaw) [inline docs/standard/linq/sort-elements.md:103]:
```suggestion
- [Sorting Data (Visual Basic)](../../visual-basic/programming-guide/concepts/linq/sorting-data.md)
```

File: docs/standard/linq/sort-elements.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+description: Wite a query that sorts its results
+ms.date: 07/20/2015
+dev_langs:
+  - "csharp"
+  - "vb"
+ms.assetid: aee6fbbc-81fd-4b3e-b40f-6ed7b3bd3fee
+---
+
+# How to sort elements (LINQ to XML)
+
```
Comment (mairaw) [inline docs/standard/linq/sort-elements.md:12]: Can you add a quick intro here?

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+# How to write queries with complex filtering (LINQ to XML)
+
+Sometimes you want to write LINQ to XML queries with complex filters. For example, you might have to find all elements that have a child element with a particular name and value. This article gives an example of writing a query with complex filtering.
+
+## Example: Find with a nested query in the `Where` clause
+
+This example shows how to find all `PurchaseOrder` elements that have:
+
+- a child `Address` element whose `Type` attribute equals "Shipping", and
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:19]:
```suggestion
- A child `Address` element whose `Type` attribute equals "Shipping".
```

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+- a child `State` element that equals "NY".
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:20]:
```suggestion
- A child `State` element that equals "NY".
```

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+ It uses a nested query in the `Where` clause, and the `Any` operator returns `true` if the collection has any elements in it. The example uses XML document [Sample XML file: Multiple purchase orders](sample-xml-file-multiple-purchase-orders.md).
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:22]:
```suggestion
It uses a nested query in the `Where` clause, and the `Any` operator returns `true` if the collection has any elements in it. The example uses XML document [Sample XML file: Multiple purchase orders](sample-xml-file-multiple-purchase-orders.md).
```

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+99505
+```
+
+## See also
+
+- <xref:System.Xml.Linq.XElement.Attribute%2A>
+- <xref:System.Xml.Linq.XContainer.Elements%2A>
+- [Projection Operations (C#)](../../csharp/programming-guide/concepts/linq/projection-operations.md)
+- [Quantifier Operations (C#)](../../csharp/programming-guide/concepts/linq/quantifier-operations.md)
+- [XML Child Axis Property](../../visual-basic/language-reference/xml-axis/xml-child-axis-property.md)
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:123]:
```suggestion
- [XML Child Axis Property (Visual Basic)](../../visual-basic/language-reference/xml-axis/xml-child-axis-property.md)
```

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+- [XML Attribute Axis Property](../../visual-basic/language-reference/xml-axis/xml-attribute-axis-property.md)
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:124]:
```suggestion
- [XML Attribute Axis Property (Visual Basic)](../../visual-basic/language-reference/xml-axis/xml-attribute-axis-property.md)
```

File: docs/standard/linq/write-queries-complex-filtering.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+- [XML Value Property](../../visual-basic/language-reference/xml-axis/xml-value-property.md)
```
Comment (mairaw) [inline docs/standard/linq/write-queries-complex-filtering.md:125]:
```suggestion
- [XML Value Property (Visual Basic)](../../visual-basic/language-reference/xml-axis/xml-value-property.md)
```

**Review by mairaw — CHANGES_REQUESTED** [review]
A few extra comments

**Comment by jameshkramer** [comment]
@mairaw @BillWagner 

Maira, thanks for the review. I wrote the intro you requested. I'll be more careful in the future when I copy links from articles (to be sure all the need link text is there).

**Review by mairaw — APPROVED** [review]
LGTM

**Pull request merged**